src/custom-elements/src/components/drawer/drawer.tsx

```diff
@@ -214,10 +214,17 @@ export class Drawer {
 
   private _documentClickHandler = (ev): void => {
     const drawerElement = this.el.querySelector('.chi-drawer');
+    const drawerCloseButton = drawerElement.querySelector('button.-close');
```

Comment:
What if the drawer contains a `button.-close` inside it. Could we use a more specific selector?

Reply:
What about:
    `const drawerCloseButton = drawerElement.querySelector('.chi-drawer__header button.-close');` ?

Reply:
That will work only for the headed drawer. But there is an option to use a drawer without header in which the button will be a direct descendant of the drawer, and that is a little bit trickier. I think the best option here is to do a tree search over `drawerElement.children` and test the classes of the child elements.

Reply:
In case of Sidenav Drawers we locate the X button like this:

const CLOSE_TRIGGER_SELECTOR = `.${CLASS_DRAWER} > .-close, .${CLASS_DRAWER} > .chi-drawer__header > .-close`;

...

_locateCloseButton() {
    const closeButtons = this._drawerElem.querySelectorAll(CLOSE_TRIGGER_SELECTOR);
    if (closeButtons) {
      return closeButtons[0];
    } else {
      return null;
    }
  }